PGF over IBC

yito88 opened the pull request:
## Describe your changes
closes #1395
CI for this PR requires Hermes: #2237

## Indicate on which release or other PRs this topic is based on
#1797

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

Fraccaman commented:
can you add the new `e2e` test to the `e2e.json` file pls

grarco commented on apps/src/lib/node/ledger/shell/governance.rs:
```diff
@@ -340,64 +347,71 @@ where
+                        &target.target,
+                        target.amount,
+                    ),
+                    PGFTarget::Ibc(target) => {
```
This actually adds support for rPGF payments over ibc which doesn't show up as a requirement in #1395. It looks good to me I just want to make sure this is intended

cwgoes replied:
ah I think having both is good

grarco reviewed (dismissed):
Looks good to me, thank you!

yito88 commented:
Rebased. I moved `ibc.rs` from `app` to `core`, which implements `transfer_over_ibc` for a protocol transaction because of conflict.
@Fraccaman @grarco Could you review it again?

grarco commented on apps/src/lib/node/ledger/shell/governance.rs:
```diff
@@ -136,6 +136,20 @@ where
                             id
                         );
 
+                        for ibc_event in
```
It looks to me like we are doing this loop to set the height for ibc events three times: once here, once in the `apply_inflation` function and once when we handle the `TxResult` coming from `dispatch_tx`. I wonder if we could do this just once at the end of `finalize_block`

grarco replied:
Nevermind, this doesn't make sense, we need to extract the events from the write log and push them to the abci response
